wit/network.wit

```diff
@@ -13,11 +13,77 @@ default interface network {
 	drop-network: func(this: network)
 
 
-
+	/// Error codes.
+	/// 
+	/// In theory, every API can return any error code.
+	/// In practice, API's typically only return the errors documented per API
+	/// combined with a couple of errors that are always possible:
+	/// - `unknown`
+	/// - `access-denied`
+	/// - `not-supported`
+	/// - `resource-limit-reached`
+	/// - `out-of-memory`
+	/// 
+	/// See each individual API for what the POSIX equivalents are. They sometimes differ per API.
 	enum error {
+		// ### GENERAL ERRORS ###
+
+		/// Unknown error
 		unknown,
-		again,
-		// TODO ...
+
+		/// POSIX equivalent: EACCES, EPERM
+		access-denied,
+
+		/// POSIX equivalent: EOPNOTSUPP
+		not-supported,
+
+		/// POSIX equivalent: EMFILE, ENFILE, ENOMEM, ENOBUFS, EAI_MEMORY
+		resource-limit-reached,
+
+		/// POSIX equivalent: ENOMEM, ENOBUFS, EAI_MEMORY
+		out-of-memory,
+
+		timeout,
+		concurrency-conflict,
```

Comment:
Could you add a documentation comment for `concurrency-conflict`? I guess it often corresponds to `EALREADY`.

Reply:
Done! d28139697d720ca8e2277cbac4ef77285afc20ea

Yes, it maps to EALREADY. (Except `listen` on BSD disagrees)